Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -687,6 +687,66 @@ public struct NonBlockingFileIO: Sendable {
         }
     }
 
+    private func createDirectory0(_ path: String, mode: NIOPOSIXFileMode) throws {
+        // Fail fast if not a directory or file exists
+        do {
+            var s = stat()
+            try Posix.stat(pathname: path, outStat: &s)
+            if (S_IFMT & s.st_mode) == S_IFDIR {
+                return
+            }
+            throw IOError(errnoCode: ENOTDIR, reason: "Not a directory")
+        } catch {
+            // ignore error
+        }
+
+        // Slow path
+        var index = path.index(before: path.endIndex)
```

Comment:
You need an early-exit here if the path is empty.

Reply:
added a check in the beginning of the function